drivers/input/touchscreen_upper.c

```diff
@@ -366,34 +350,40 @@ static int touch_poll(FAR struct file *filep,
 void touch_event(FAR void *priv, FAR const struct touch_sample_s *sample)
 {
   FAR struct touch_upperhalf_s *upper = priv;
+  FAR struct touch_openpriv_s  *openpriv;
   int semcount;
 
   if (nxsem_wait(&upper->exclsem) < 0)
     {
       return;
     }
 
-  circbuf_overwrite(&upper->buffer, sample,
-                    SIZEOF_TOUCH_SAMPLE_S(sample->npoints));
-  touch_notify(upper, POLLIN);
-  nxsem_get_value(&upper->buffersem, &semcount);
-  if (semcount < 1)
+  list_for_every_entry(&upper->head, openpriv, struct touch_openpriv_s, node)
     {
-      nxsem_post(&upper->buffersem);
+      circbuf_overwrite(&openpriv->circbuf, sample,
+                        SIZEOF_TOUCH_SAMPLE_S(sample->npoints));
+
+      nxsem_get_value(&openpriv->waitsem, &semcount);
+      if (semcount < 1)
```

Comment:
why do we need check semaphore count value? can we `nxsem_post(&openpriv->waitsem);` without a check?

Reply:
We don't seem to need it.
When the semaphore count value is greater than zero and the buff is empty, nxsem_wait will consume the count value in the cycle.

Reply:
it could avoid poll/read wake up multiple time but the fifo is empty.